File: codeforces/86.cards_for_friends_[1472A]/cards_for_friends.py
# Author: José Rodolfo (jric2002)
import math
t = int(input())
while (t):
  w, h, n = list(map(int, input().split(" ")))
  x = 0
  while ((w % 2) == 0):
    w = w / 2
    x += 1
  while ((h % 2) == 0):
    h = h / 2
    x += 1
  if (math.pow(2, x) >= n):
    print("YES")
  else:
    print("NO")
  t -= 1

File: codeforces/86.cards_for_friends_[1472A]/cards_for_friends.cpp
#include <iostream>
#include <cmath>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, w, h, x;
  unsigned int n;
  cin >> t;
  while (t--) {
    cin >> w >> h >> n;
    x = 0;
    while ((w % 2) == 0) {
      w = w / 2;
      x++;
    }
    while ((h % 2) == 0) {
      h = h / 2;
      x++;
    }
    if (pow(2, x) >= n) {
      cout << "YES" << endl;
    }
    else {
      cout << "NO" << endl;
    }
  }
  return 0;
}